### PR TITLE
WIP: cmd/dbconvert: implement pebble<->leveldb conversion

### PR DESCRIPTION
**Pull request by holiman**
This PR implements a converter between leveldb and pebble (both directions).

Going from leveldb to pebble: 
```
[user@work go-ethereum]$ go run ./cmd/dbconvert/ to-pebble --dbdir /home/user/.ethereum/yolo-v3/geth/chaindata/
INFO [04-21|20:00:37.397] Allocated cache and file handles         database=/home/user/.ethereum/yolo-v3/geth/chaindata cache=4.00GiB handles=4096
INFO [04-21|20:00:38.393] Using LevelDB as the backing database 
INFO [04-21|20:00:38.393] Allocated cache and file handles         database=/home/user/.ethereum/yolo-v3/geth/.tempdir cache=4.00GiB handles=4096
INFO [04-21|20:00:39.409] Converting database                      elapsed=2.571895ms items=1 size=106.00B
INFO [04-21|20:00:43.036] Releasing iterator, flushing deletes 
INFO [04-21|20:00:44.239] Converted database                       elapsed=4.832718904s items=931,711 size=61.22MiB
INFO [04-21|20:00:44.289] Moving ancients                          from=/home/user/.ethereum/yolo-v3/geth/chaindata/ancient to=/home/user/.ethereum/yolo-v3/geth/.tempdir/ancient
INFO [04-21|20:00:44.289] Deleting old directory                   dir=/home/user/.ethereum/yolo-v3/geth/chaindata
INFO [04-21|20:00:44.301] Moving new directory                     from=/home/user/.ethereum/yolo-v3/geth/.tempdir to=/home/user/.ethereum/yolo-v3/geth/chaindata
```
And back again:
```
[user@work go-ethereum]$ go run ./cmd/dbconvert/ to-ldb --dbdir /home/user/.ethereum/yolo-v3/geth/chaindata/
INFO [04-21|20:01:56.776] Allocated cache and file handles         database=/home/user/.ethereum/yolo-v3/geth/chaindata cache=4.00GiB handles=4096
INFO [04-21|20:01:57.869] Allocated cache and file handles         database=/home/user/.ethereum/yolo-v3/geth/.tempdir cache=4.00GiB handles=4096
INFO [04-21|20:01:58.233] Using LevelDB as the backing database 
INFO [04-21|20:01:58.233] Converting database                      elapsed="5.87µs" items=1 size=106.00B
INFO [04-21|20:01:59.369] Releasing iterator, flushing deletes 
INFO [04-21|20:02:01.136] Converted database                       elapsed=2.903054092s items=931,711 size=61.22MiB
INFO [04-21|20:02:01.164] Moving ancients                          from=/home/user/.ethereum/yolo-v3/geth/chaindata/ancient to=/home/user/.ethereum/yolo-v3/geth/.tempdir/ancient
INFO [04-21|20:02:01.164] Deleting old directory                   dir=/home/user/.ethereum/yolo-v3/geth/chaindata
INFO [04-21|20:02:01.198] Moving new directory                     from=/home/user/.ethereum/yolo-v3/geth/.tempdir to=/home/user/.ethereum/yolo-v3/geth/chaindata
```

It copies all db-data, and removes them from the original db (most user probably do not want to have double storage capacity requirements to do the conversion). It also moves the `ancient`-folder, so everything "should just work". 

However, there todos
- It should gracefully handle aborts, if user presses `ctrl-c`, and be able to resume later on. 
- There are probably more optimal ways to decide when to flush deletes and re-init the iterator, etc, to maximize the speed.

### CODE REVIEW

**Comment by karalabe** [comment]
The catch is that this will not only need double space requirements during conversion, but most probably like 3x. When deleting stuff from leveldb, it does not just remove it from storage, rather it *appends* deletion markers at the lower levels. Those need to get compacted eventually all the way to level 7 until the data is truly removed. That however is a very significant delay which usually means the db bloats up first, before eventually coming back down.

I guess if you can somehow force compactions as you're going along it might work, but this is definitely a significant performance consideration. Also, a resync does have the advantage of removing all the accumulated junk. Also path based storage will need a resync anyway.

TL;DR: Whilst nice, I'm unsure if the ever so slight user nicety outweighs the complexity and all the ways it can go wrong.

**Comment by holiman** [comment]
> TL;DR: Whilst nice, I'm unsure if the ever so slight user nicety outweighs the complexity and all the ways it can go wrong.

Yeah, I agree. This is one of those things that are "good to have on a branch", in case it's needed at some point, but it doesn't have to be part of the `master` code
